src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentLoweringPass.cs

```diff
@@ -115,9 +118,25 @@ public override void VisitTagHelper(TagHelperIntermediateNode node)
 
             public override void VisitTagHelperBody(TagHelperBodyIntermediateNode node)
             {
-                for (var i = 0; i < node.Children.Count; i++)
+                // Wrap the component's children in a ChildContent node
+                if (node.Children.Count > 0)
                 {
-                    _children.Add(node.Children[i]);
+                    var childContent = new ComponentChildContentIntermediateNode();
+                    var childContentAttribute = _component.Component.BoundAttributes
+                        .Where(a => a.Name == BlazorApi.RenderTreeBuilder.ChildContent)
+                        .FirstOrDefault();
+                    if (childContentAttribute != null)
+                    {
+                        // This component accepts child content explicitly.
+                        childContent.Attribute = childContentAttribute;
```

Comment:
The use of the term "attribute" here is a bit quirky. Is this a Razorism based on the fact that normally, parameters are represented as attributes? In this case we're dealing with something like a "parameter" or a "child element" in most cases rather than an HTML attribute, right?

Reply:
This is indeed a Razor-ism. The `childContent.Attribute` is a property of type `BoundAttributeDescriptor` defined in Razor. We're using the existing OM for tag helpers to describe components.